mobilecoind/strategies/burn.py

```diff
@@ -0,0 +1,228 @@
+            empty_pb2.Empty()
+        ).network_highest_block_index
+
+    def get_utxos(self, monitor_id, token_id, subaddress_index=0):
+        return self.stub.GetUnspentTxOutList(
+            GetUnspentTxOutListRequest(
+                monitor_id=monitor_id,
+                subaddress_index=subaddress_index,
+                token_id=token_id,
+            )
+        ).output_list
+
+    def generate_burn_redemption_tx(
+        self,
+        monitor_id,
+        input_list,
+        burn_amount,
+        fee,
+        token_id,
+        change_subaddress=0,
+        tombstone=None,
+        redemption_memo=None,
+        enable_destination_memo=False,
+    ):
+        if tombstone is None:
+            tombstone = self.get_network_block_index() + 10
+
+        request = GenerateBurnRedemptionTxRequest(
+            sender_monitor_id=monitor_id,
+            change_subaddress=change_subaddress,
+            input_list=input_list,
+            burn_amount=burn_amount,
+            fee=fee,
+            tombstone=tombstone,
+            token_id=token_id,
+            redemption_memo=redemption_memo,
+            enable_destination_memo=enable_destination_memo,
+        )
+        return self.stub.GenerateBurnRedemptionTx(request).tx_proposal
+
+    def submit_tx(self, tx_proposal):
+        return self.stub.SubmitTx(
+            SubmitTxRequest(
+                tx_proposal=tx_proposal,
+            )
+        )
+
+    def wait_for_monitor_to_sync(self, monitor_id, poll_interval=0.1):
+        while True:
+            network_block_index = self.get_network_block_index()
+            status = self.stub.GetMonitorStatus(
+                GetMonitorStatusRequest(
+                    monitor_id=monitor_id
+                )
+            ).status
+            if status.next_block >= network_block_index:
+                break
+            time.sleep(poll_interval)
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    logging.debug(args)
+
+    client = MobilecoindClient(f"{args.mobilecoind_host}:{args.mobilecoind_port}")
+    monitor = client.new_monitor_from_keyfile(args.key)
+
+    client.wait_for_monitor_to_sync(monitor)
+
+    balance = client.get_balance(monitor, args.token_id)
+
+    if balance < args.value + args.fee:
+        raise Exception(f"Insufficient balance: {balance} < {args.value} + {args.fee}")
+
+    utxos = client.get_utxos(monitor, args.token_id)
+    tx_proposal = client.generate_burn_redemption_tx(
+        monitor,
+        utxos,
+        args.value,
+        args.fee,
+        args.token_id,
+    )
+    logging.info("Constructed burn transaction")
+    logging.debug(tx_proposal)
+
+    # Search for the burn txout. mobilecoind will create one "outlay" which is
+    # the burn, and there is a map in the TxProposal that tells you which TxOut it is.
+    if len(tx_proposal.outlay_index_to_tx_out_index) != 1:
+        logging.warning(f"Unexpected number of outlays: {tx_proposal.outlay_index_to_tx_out_index}")
```

Comment:
Maybe this should cause the script to fail? e.g. raise an exception. It's unlikely anyone will ever look at the logs of a CI script unless it fails